code/modules/antagonists/eldritch_cult/eldritch_antag.dm

```diff
@@ -52,11 +54,16 @@
 	if(!silent)
 		to_chat(owner.current, "<span class='userdanger'>Your mind begins to flare as the otherwordly knowledge escapes your grasp!</span>")
 		owner.current.log_message("has become a non-heretic", LOG_ATTACK, color="#960000")
-	GLOB.reality_smash_track.RemoveMind(owner)
 	STOP_PROCESSING(SSprocessing,src)
 
+	on_death()
 	return ..()
 
+/datum/antagonist/heretic/proc/on_death()
+
+	for(var/X in researched_knowledge)
+		var/datum/eldritch_knowledge/EK = researched_knowledge[X]
+		EK.on_death(owner.current)
```

Comment:
`on_death` part is from tgstation/tgstation#54252

Reply:
Thanks. I missed this one.